packages/EchoFramework/src/hooks/useHas3dAccess.ts

```
@@ -0,0 +1,35 @@
+import { BaseError, useInstCode } from '@equinor/echo-core';
+import { useEffect, useState } from 'react';
+import { check3dPermissionForPlantCode, check3dPermissionOnInstallation } from '../utils/plantInfo';
+/**
+ * Hook for checking if user has access to models for a given asset
+ *
+ * @export
+ * @param {string} [plantCode] asset to check access for, if not supplied instCode from user settings
+ * @param {((error: Error | BaseError | unknown) => void)} [handleError] method that will handle if an error occurred
+ * @return {*}  {boolean} returns wether or not the user have access
+ */
+export default function useHas3dAccess(
+    plantCode?: string,
+    handleError?: (error: Error | BaseError | unknown) => void
```

handleError should not be a dependency here, it makes a simple function difficult to use. And optional means we silently ignore the error, it's never reported to AppInsight

---

If not as a dependency then how? 
If nothing then the error will be completely ignored, and thats probably not wanted
also this is a hook so its not like it can be wrapped in a try catch and catch the error from the outside